File: leetcode/problems/238_product_of_array_except_self.go
func productExceptSelf(nums []int) []int {
	N := len(nums)
	ret := make([]int, N)
	for idx, _ := range nums {
		ret[idx] = 1
	}

	multiplier := 1
	for i := 1; i < N; i++ {
		multiplier *= nums[i-1]
		ret[i] *= multiplier
	}

	multiplier = 1
	for i := N - 2; i > -1; i-- {
		multiplier *= nums[i+1]
		ret[i] *= multiplier
	}
	return ret
}

File: leetcode/problems/238_product_of_array_except_self.py
from typing import List


class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        N = len(nums)
        result = [1] * N

        multiplier = 1
        for i in range(1, N):
            multiplier *= nums[i-1]
            result[i] *= multiplier
        
        multiplier = 1
        for i in range(N-2, -1, -1):
            multiplier *= nums[i+1]
            result[i] *= multiplier


        return result
